shared/lib/tasks.py

```diff
@@ -33,7 +48,8 @@ def execute(self):
 		self.duration = time.time()-self.last
 		self.last = self.next
 		if self.repeated:
-			self.next = time.time() + self.timeout
+			self.next = time.time() + self.next_timeout
+			self.next_timeout = self.timeout
```

Comment:
See my comment above. this makes sure that self.next_timeout is only used once, and self.timeout is used for all further executions.